sorting/quick_sort.cpp

```diff
@@ -1,16 +1,42 @@
-/*
+/**
  *
  *      copyright The Algorithms
  *      Author - 
  *      Correction - ayaankhan98
  *
+ * Implementation Details -
+ *      Quick Sort is a divide and conquer algorithm. It picks and element as
+ *      pivot and partition the given array around the picked pivot. There
+ *      are many different versions of quickSort that pick pivot in different
+ *      ways.
+ *
+ *      1. Always pick the first element as pivot
+ *      2. Always pick the last element as pivot (implemented below)
+ *      3. Pick a random element as pivot
+ *      4. Pick median as pivot
+ *
+ *      The key process in quickSort is partition(). Target of partition is,
+ *      given an array and an element x(say) of array as pivot, put x at it's
+ *      correct position in sorted array and put all smaller elements (samller
+ *      than x) before x, and put all greater elements (greater than x) after
+ *      x. All this should be done in linear time
+ *
  */
 
 #include <cstdlib>
 #include <iostream>
 
+/**
+ *      This function takes last element as pivot, places  
+ *      the pivot element at its correct position in sorted  
+ *      array, and places all smaller (smaller than pivot)  
+ *      to left of pivot and all greater elements to right  
+ *      of pivot 
+ *
+ */
+
 int partition(int arr[], int low, int high) {
-    int pivot = arr[high];    // pivot
+    int pivot = arr[high];    // taking the last element as pivot
```

Comment:
Taking the extreme elements as pivot lands up the algorithm in the worst case scenario in case where data is sorted. So, if you choose your pivot as the middle element or try out a random element that will land up in the average case complexity.